contracts/ActivePool.sol

```diff
@@ -111,6 +125,13 @@ contract ActivePool is
 
 		assetsBalance[_asset] = assetsBalance[_asset].sub(_amount);
 
+		if (assetsStaked[_asset] > assetsBalance[_asset]) {
+			uint256 shortage = assetsStaked[_asset] - assetsBalance[_asset];
+			assetsStaked[_asset] = assetsStaked[_asset].sub(shortage);
```

Comment:
`assetsStaked[_asset] -= shortage;` gas friendly

Also you do a lot of storage reading here. If you use a data more than once, you are better to use the stack memory.

i.e:
```
uint256 totalAssets= assetsBalance[_asset] -= _amount;
uint256 stakedAssets = assetsStaked[_asset];

if( stakedAssets > totalAssets){
 uint256 short = stakedAssets - totalAssets;
 assetsStaked[_asset] -= shortage
[....]
}


```